nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,467 @@
+    ) -> pd.DataFrame:
+        """Add manifest rewards column to the status DataFrame."""
+        manifest_col = status_col_dict["manifest"]
+        doughnut_cols = status_col_dict["doughnut"]
+        bagel_cols = status_col_dict["bagel"]
+
+        status_count_cols = [manifest_col] + doughnut_cols + bagel_cols
+        status_df[status_count_cols] = (
+            status_df[status_count_cols].fillna(0).astype(int)
+        )
+
+        for row_obj in status_df.iterrows():
+            row = row_obj[1]
+            reward_emojis = []
+            # clipboard emoji if some task (curation or processing) has been
+            # performed on participants in manifest else sad cat emoji
+            if row[manifest_col] > 0:
+                if (row[doughnut_cols] == 0).all() & (row[bagel_cols] == 0).all():
+                    reward_emojis.append(sad_cat_emoji)
+                else:
+                    reward_emojis.append(clipboard_emoji)
+
+            status_df.loc[row_obj[0], reward_col_name] = " ".join(reward_emojis)
+
+        return status_df
+
+    def _add_doughnut_rewards(
+        self, status_df: pd.DataFrame, status_col_dict: dict, reward_col_name: str
+    ) -> pd.DataFrame:
+        """Add doughnut rewards column to the status DataFrame."""
+        manifest_col = status_col_dict["manifest"]
+        doughnut_cols = status_col_dict["doughnut"]
+        bids_col = status_col_dict["bids"]
+        bagel_cols = status_col_dict["bagel"]
+
+        status_count_cols = [manifest_col] + doughnut_cols + bagel_cols
+        status_df[status_count_cols] = (
+            status_df[status_count_cols].fillna(0).astype(int)
+        )
+
+        for row_obj in status_df.iterrows():
+            row = row_obj[1]
+            reward_emojis = []
+            # doughnut emoji if at least one participant has been bidsified
+            # can't rely on other columns if we begin with BIDS dataset
+            if (row[bids_col] > 0) & (row[bids_col] <= row[manifest_col]):
+                reward_emojis.append(doughnut_emoji)
+            # sparkles emoji if all participants have been bidsified
+            if row[bids_col] == row[manifest_col]:
+                reward_emojis.append(sparkles_emoji)
+            # broom emoji if at least one participant has started curation
+            elif (row[doughnut_cols] > 0).any():
+                reward_emojis.append(broom_emoji)
+
+            # Additional rewards based on quantity
+            # if row[bids_col] > high_throughput:
+            #     reward_emojis.append(bento_box_emoji)
+            # elif row[bids_col] > medium_throughput:
+            #     reward_emojis.append(sushi_emoji)
+
+            status_df.loc[row_obj[0], reward_col_name] = " ".join(reward_emojis)
+
+        return status_df
+
+    def _add_bagel_rewards(
+        self, status_df: pd.DataFrame, status_col_dict: dict, reward_col_name: str
+    ) -> pd.DataFrame:
+        """Add begal rewards column to the status DataFrame."""
```

Comment:
```suggestion
        """Add bagel rewards column to the status DataFrame."""
```

I'm surprised Codespell didn't catch that...